### PR TITLE
Refactorings

### PR DESCRIPTION
**Pull request by adamnemecek**

### CODE REVIEW

File: Sources/UInt128.swift [hunk 1]
```diff
@@ -662,20 +667,11 @@ extension UInt128 : ExpressibleByStringLiteral {
     // MARK: Type Methods
 
     internal static func _valueFromString(_ value: String) -> UInt128? {
-        let radix = UInt128._determineRadixFromString(value)
+        let radix = value.radix
```
Comment (Jitsusama) [inline Sources/UInt128.swift]: I’m having a hard time understanding what value this refactoring brings. Could you explain the rationale behind replacing a local method with a private extension?

File: Sources/UInt128.swift [hunk 2]
```diff
@@ -429,19 +429,15 @@ extension UInt128 : BinaryInteger {
     // MARK: Type Methods
 
     public static func /(lhs: UInt128, rhs: UInt128) -> UInt128 {
-        let result = lhs.dividedReportingOverflow(by: rhs)
-
-        return result.partialValue
+        return lhs.dividedReportingOverflow(by: rhs).partialValue
```
Comment (Jitsusama) [inline Sources/UInt128.swift]: I would prefer to not have the train of calls. It’s mostly for reasons matching my own escetic, but I can’t see any advantage to doing it differently then what I already have. Ditto for the next 3 places you did this.

File: Sources/UInt128.swift [hunk 3]
```diff
@@ -603,18 +597,18 @@ extension UInt128 : CustomStringConvertible {
     /// - returns:
     ///     String representation of the stored UInt128 value.
     internal func _valueToString(radix: Int = 10, uppercase: Bool = true) -> String {
-        precondition(radix > 1 && radix < 37, "radix must be within the range of 2-36.")
+        precondition((2...36).contains(radix), "radix must be within the range of 2-36.")
```
Comment (Jitsusama) [inline Sources/UInt128.swift]: Any particular reason for changing the logic here?

File: Sources/UInt128.swift (hunk 1 above)
Comment (adamnemecek) [reply Sources/UInt128.swift]: because radix is fundamentally a property of the string.

File: Sources/UInt128.swift (hunk 3 above)
Comment (adamnemecek) [reply Sources/UInt128.swift]: logic is the same, it's just that the range communicates the intent clearer.

Comment (Jitsusama) [reply Sources/UInt128.swift]: I guess I can see how it is a bit clearer, so sure, why not

File: Sources/UInt128.swift (hunk 1 above)
Comment (Jitsusama) [reply Sources/UInt128.swift]: I see where you're coming from, but I believe I'd like to keep the structure as it was.

**Pull request closed without merging**